src/ecs/commands/runCommandInContainer.ts

```diff
@@ -87,3 +116,35 @@ export async function runCommandInContainer(
         status?.dispose()
     }
 }
+/**
+ * Attempts to verify if the task role provided has the required SSM permissions to run ECS Exec
+ * @param taskRoleArn
+ * @param iamClient
+ * @returns True if the permissions are missing from the Task role. False when all required permissions were found. Undefined when the task role is missing or the 'simulatePrincipalPolicy' call was unsuccessful.
+ */
+export async function isMissingRequiredPermissions(
+    taskRoleArn: string | undefined,
+    iamClient: IamClient
+): Promise<boolean | undefined> {
+    if (taskRoleArn === undefined) {
+        return undefined
+    }
+    try {
+        const permissionResponse = await iamClient.simulatePrincipalPolicy({
+            PolicySourceArn: taskRoleArn,
+            ActionNames: REQUIRED_SSM_PERMISSIONS,
+        })
+        if (!permissionResponse || !permissionResponse.EvaluationResults) {
+            return undefined
+        }
+        for (const evalResult of permissionResponse.EvaluationResults) {
+            if (evalResult.EvalDecision !== 'allowed') {
```

Comment:
None of this logic looks specific to ECS. Could add `hasRolePermissions` on `iamClient`, then ECS exec call would be:

```
await iamClient.hasRolePermissions({
            PolicySourceArn: taskRoleArn,
            ActionNames: REQUIRED_SSM_PERMISSIONS,
        })
```